### PR TITLE
Convert enrollment tests to react testing library

### PR DESCRIPTION
**Pull request by mmazanec22**
Wow this is so much cleaner, and I love that the snapshots are readable!

### CODE REVIEW

File: src/Hedwig/ClientApp/src/containers/Enrollment/Edit/EnrollmentEdit.test.tsx [hunk 1]
```diff
@@ -45,29 +44,24 @@ describe('EnrollmentEdit', () => {
...
+
+			// Save without entering any data
+			fireEvent.click(getByText('Save'));
+			// TODO: also try with keyboard event?
+
+			const firstNameErr = await waitForElement(() =>
+				getByText('This information is required for enrollment')
+			);
+
+			expect(firstNameErr).toBeInTheDOM();
```
Comment (cailyncodes) [inline src/Hedwig/ClientApp/src/containers/Enrollment/Edit/EnrollmentEdit.test.tsx]: Some bad naming conventions on behalf of `jest-dom`: https://github.com/testing-library/jest-dom#tobeinthedom

Tl;dr -- we should use "toBeInTheDocument or toContainElement."

File: src/Hedwig/ClientApp/src/containers/Enrollment/Edit/EnrollmentEdit.test.tsx [hunk 2]
```diff
@@ -115,20 +97,15 @@ describe('EnrollmentEdit', () => {
...
-				.find('EnrollmentEdit')
-				.dive()
-				.find('Form')
-				.dive()
-				.find('Alert')
-				.props().type;
-			expect(alertPropsType).toBe('info');
+			const infoAlert = getByText('Income information is required to enroll a child in a CDC funded space. You will not be able to assign this child to a funding space without this information.');
+
+			expect(infoAlert).toBeInTheDOM();
```
Comment (cailyncodes) [inline src/Hedwig/ClientApp/src/containers/Enrollment/Edit/EnrollmentEdit.test.tsx]: Here too

File: src/Hedwig/ClientApp/src/containers/Enrollment/Edit/EnrollmentEdit.test.tsx [hunk 3]
```diff
@@ -80,28 +74,16 @@ describe('EnrollmentEdit', () => {
...
-			const addressErr = formContent
-				.find('FieldSet#family-address')
-				.dive()
-				.find('FormStatus')
-				.props().type;
-			expect(addressErr).toBe('warning');
+
+			const addressErr = getByRole('status');
+
+			expect(addressErr.className).toBe('usa-warning-message');
```
Comment (cailyncodes) [inline src/Hedwig/ClientApp/src/containers/Enrollment/Edit/EnrollmentEdit.test.tsx]: Perhaps we want to do a `className.classList.includes()` so multiple classes can be added to the element?
